docs/DeveloperGuide.md

```diff
@@ -252,51 +252,299 @@ b. Require developers to download those libraries manually (this creates extra w
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `Application` and the **Actor** is the `user`, unless specified otherwise)
+
+#### Use case: Usage Instructions
+**MSS**
+
+1. User want to view all the available commands in the application.
+2. Application show a list of available commands and instructions for all commands.
+    Use Case ends
+
+#### Use case : Add task
+
+**MSS**
+
+1. User request to add new tasks with its details.
+2. Application add the task with specified details.  
+3. Application display successful message.
+Use case ends.  
+
+**Extensions**  
+1a. User specify with start date and end date.
+
+> 1a1. User specify invalid date format.
+  Application shows an error message.
+  Use case ends.
+  
+1b. User specify with deadline.  
+
+> 1b1. User specify invalid date deadline format.
+   Application shows an error message.
+   Use case ends.
 
-#### Use case: Delete person
+#### Use case : Find tasks with specific keyword
 
 **MSS**
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
+1. User request to find task with specific keyword.
+2. Application show the list of task that its names & tag contain the keyword.
+Use case ends.  
+
+**Extensions**  
+2a. The list is empty.
+
+> 2a1. Application show an error message.
+   Use case ends.
+
+#### Use case : List all task
+
+**MSS**
+
+1. User request to list all the tasks.
+2. Application show the list of task with respective details.  
+Use case ends.  
+
+**Extensions**  
+2a. The list is empty.
+
+> 2a1. Application show error message.
+   Use case ends.
+   
+#### Use case: Delete task
+
+**MSS**
+
+1. User requests a list of tasks or find task with keyword.
+2. Application shows a list of tasks.
+3. User requests to delete a specific task in the list.
+4. Application request confirmation from the user.
+5. User confirm the confirmation.
+4. Application deletes the task. <br>
+Use case ends.  
+
+**Extensions**  
+
+2a. The list is empty.
+
+> Use case ends.
+
+3a. The given index is invalid.
+
+> 3a1. Application shows an error message.  
+  Use case ends.
+
+#### Use case: edit task
+
+**MSS**
+
+1. User requests a list of tasks or find task with keyword.
+2. Application shows a list of tasks.
+3. User requests to edit a specific task in the list.
+4. Application edit the task. <br>
+Use case ends.  
+
+**Extensions**  
+
+2a. The list is empty.
+
+> Use case ends.
+
+3a. The given index is invalid.
+
+> 3a1. Application shows an error message.  
+  Use case ends.
+
+#### Use case : Undo command
+
+**MSS**
+1. User request to undo command by a specific number.
+2. Application request confirmation from the user.
+3. User confirm the confirmation.
+2. Application undo the command repeatedly based on the given number.
```

Comment:
mmmm..should we really request confirmation every time? think about about UX for this